pkgs/development/python-modules/defusedxml/default.nix

```diff
@@ -1,11 +1,25 @@
-{ buildPythonPackage, fetchPypi }:
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+}:
 
 buildPythonPackage rec {
   pname = "defusedxml";
-  version = "0.6.0";
+  version = "0.7.0";
+  disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "f684034d135af4c6cbb949b8a4d2ed61634515257a67299e5f940fbaa34377f5";
+    sha256 = "183fz8xwclhkirwpvpldyypn47r8lgzfz2mk9jgyg7b37jg5vcc6";
+  };
+
+  pythonImportsCheck = [ "defusedxml" ];
```

Comment:
Enable tests by passing `tests.py` to the pytestCheckHook.

python3Packages.defusedxml

Reply:
Not required https://github.com/tiran/defusedxml/blob/master/tox.ini#L42

Reply:
```diff
diff --git a/pkgs/development/python-modules/defusedxml/default.nix b/pkgs/development/python-modules/defusedxml/default.nix
index 3abdd0fb792..ef1ba91d45c 100644
--- a/pkgs/development/python-modules/defusedxml/default.nix
+++ b/pkgs/development/python-modules/defusedxml/default.nix
@@ -2,6 +2,7 @@
 , buildPythonPackage
 , fetchPypi
 , pythonOlder
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
@@ -14,6 +15,8 @@ buildPythonPackage rec {
     sha256 = "183fz8xwclhkirwpvpldyypn47r8lgzfz2mk9jgyg7b37jg5vcc6";
   };
 
+  checkInputs = [ pytestCheckHook ];
+
   pythonImportsCheck = [ "defusedxml" ];
 
   meta = with lib; {
```

```
=========================== short test summary info ============================
FAILED tests.py::BaseTests::test_allow_expansion - AttributeError: 'BaseTests...
FAILED tests.py::BaseTests::test_dtd_forbidden - AttributeError: 'BaseTests' ...
FAILED tests.py::BaseTests::test_dtd_with_external_ref - AttributeError: 'Bas...
FAILED tests.py::BaseTests::test_entities_forbidden - AttributeError: 'BaseTe...
FAILED tests.py::BaseTests::test_entity_cycle - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_external_file_ref - AttributeError: 'BaseTes...
FAILED tests.py::BaseTests::test_external_ref - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_simple_parse - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_simple_parse_ns - AttributeError: 'BaseTests...
FAILED tests.py::TestDefusedElementTree::test_html_arg - AssertionError: Depr...
FAILED tests.py::TestDefusedcElementTree::test_html_arg - AssertionError: Dep...
FAILED tests.py::TestDefusedLxml::test_allow_expansion - TypeError: catching ...
FAILED tests.py::TestDefusedLxml::test_dtd_forbidden - TypeError: catching cl...
FAILED tests.py::TestDefusedLxml::test_dtd_with_external_ref - TypeError: cat...
FAILED tests.py::TestDefusedLxml::test_entities_forbidden - TypeError: catchi...
FAILED tests.py::TestDefusedLxml::test_entity_cycle - TypeError: assertRaises...
FAILED tests.py::TestDefusedLxml::test_lxml_warnings - AssertionError: None i...
FAILED tests.py::TestDefusedLxml::test_restricted_element1 - TypeError: catch...
FAILED tests.py::TestDefusedLxml::test_restricted_element2 - TypeError: catch...
FAILED tests.py::TestDefusedLxml::test_simple_parse - TypeError: catching cla...
FAILED tests.py::TestDefusedLxml::test_simple_parse_ns - TypeError: catching ...
FAILED tests.py::TestDefusedLxml::test_xpath_injection - AttributeError: 'Non...
================== 22 failed, 67 passed, 2 warnings in 0.91s ===================
```

Reply:
and fails with lots of type errors.

Reply:
Uh, I'm a bit baffled why you merge this anyway. The tests are working upstream.

https://github.com/tiran/defusedxml/runs/2029830851

Reply:
@SuperSandro2000, made the requested changes in a new PR https://github.com/NixOS/nixpkgs/pull/115341